### ISSUE TITLE
KUBE-DNS Container Creating

### ISSUE THREAD

**Issue by infinitydon** [weaveworks/weave-kube#35]
Hi, 

I am running kubernetes 1.4 on centos 7 but every time I initialize weave on the cluster, kube-dns gets stuck at ContainerCreating:

NAMESPACE     NAME                                 READY     STATUS              RESTARTS   AGE
kube-system   etcd-k8smaster1                      1/1       Running             2          4m
kube-system   kube-apiserver-k8smaster1            1/1       Running             3          4m
kube-system   kube-controller-manager-k8smaster1   1/1       Running             2          4m
kube-system   kube-discovery-982812725-nlr9k       1/1       Running             0          4m
kube-system   kube-dns-2247936740-nq188            0/3            ContainerCreating   0          4m
kube-system   kube-proxy-amd64-d3vix               1/1       Running             0          4m
kube-system   kube-proxy-amd64-t6c7v               1/1       Running             0          3m
kube-system   kube-scheduler-k8smaster1            1/1       Running             2          4m
kube-system   weave-net-334ye                      2/2       Running             0          2m
kube-system   weave-net-rq0cc                      1/2       CrashLoopBackOff    4          2m

I have tried installing from scratch but still the same...

Anyone else facing this issue?

**Comment by awh** [comment]
Hi @infinitydon 

Thanks for the report. Can you attach the logs from the weave net pod that is in CrashLoopBackOff (weave-net-rq0cc) ?

**Comment by infinitydon** [comment]
Hi awh,

[root@k8smaster1 ~]# kubectl logs weave-net-rq0cc weave --namespace=kube-system
Error from server: Get https://k8sslave1:10250/containerLogs/kube-system/weave-net-rq0cc/weave: dial tcp: lookup k8sslave1 on 10.1.224.1:53: server misbehaving
[root@k8smaster1 ~]# 
[root@k8smaster1 ~]# 
[root@k8smaster1 ~]# 
[root@k8smaster1 ~]# kubectl logs weave-net-rq0cc weave-npc --namespace=kube-system
Error from server: Get https://k8sslave1:10250/containerLogs/kube-system/weave-net-rq0cc/weave-npc: dial tcp: lookup k8sslave1 on 10.1.224.1:53: server misbehaving
[root@k8smaster1 ~]# 

Is that the logs you need?

**Comment by infinitydon** [comment]
I created an entry for k8sslave1 in /etc/hosts on the master.... But now I am getting the following errors from the logs:

2016-10-11 18:21:50.218504 I | error contacting APIServer: Get https://100.64.0.1:443/api/v1/nodes: dial tcp 100.64.0.1:443: getsockopt: connection refused; trying with blank env vars
2016-10-11 18:21:50.219754 I | error contacting APIServer: Get http://localhost:8080/api: dial tcp: lookup localhost: no such host
Failed to get peers

I don't have any info about the ip address 100.64.0.1.

But telneting to port 443 on the IP shows that I can access it:

[root@k8smaster1 ~]# telnet 100.64.0.1 443
Trying 100.64.0.1...
Connected to 100.64.0.1.
Escape character is '^]'.

c

**Comment by infinitydon** [comment]
I think I got this sorted out, below are the steps I took:

1.) I don't a full DNS structure to resolve the master and slave nodes hostname, so I did this manually via /etc/hosts file.

2.) VirtualBox was giving the same IP to the master and slave (NAT Network), so I created another NAT Network and attached the VMs to it. Then I re-initialized the kubernetes cluster again and the kube-dns  started to run for the first in 2 weeks :)

Thanks awh!
